cap_client/cap_api.py

```diff
@@ -254,14 +254,20 @@ def patch(self, pid=None, filename=''):
     ##############
     def get_field(self, pid, field=None):
         """Return metadata on analysis."""
-        dct = self._make_request(url=urljoin('deposits/', pid),
-                                 headers={
-                                     'Accept': 'application/basic+json'
-        })['metadata']
+        resp_meta = self._make_request(
+            url=urljoin('deposits/', pid),
+            headers={'Accept': 'application/basic+json'}
+        )['metadata']
+
         fields = field.split('.') if field else []
         for x in fields:
-            dct = dct[x or int(x)]
-        return dct
+            try:
+                idx = int(x)
+            except ValueError:
+                idx = x
+            resp_meta = resp_meta[idx]
+
+        return resp_meta
```

Comment:
If you try to give an index that is out of range, client crashes with `unexpected error` message. It's kinda expected error ^^  so let's return to the user nicer message. You just need to add an extra block to catch an `IndexError` and return proper message.